src/lib.rs

```diff
@@ -424,15 +640,189 @@ impl TestHarness {
     pub fn create_block(&mut self) {
         self.poll_events();
 
-        let tx_hashes: Vec<_> = {
-            let txs = self.state().transactions().read().expect(
-                "Cannot read transactions from node",
-            );
-            txs.keys().cloned().collect()
-        };
+        let tx_hashes: Vec<_> = self.mempool().keys().cloned().collect();
 
         self.do_create_block(&tx_hashes);
     }
+
+    /// Creates a chain of blocks until a given height.
+    pub fn create_blocks_until(&mut self, height: Height) {
+        while self.height() < height {
+            self.create_block();
+        }
+    }
+
+    /// Returns the current height of the blockchain.
+    pub fn height(&self) -> Height {
+        self.state().height().next()
+    }
+
+    /// Returns the test node memory pool handle.
+    pub fn mempool(&self) -> RwLockReadGuard<BTreeMap<crypto::Hash, Box<Transaction>>> {
+        self.mempool.read().expect(
+            "Can't read transactions from the mempool.",
+        )
+    }
+
+    /// Returns the leader on the current height. At the moment first validator.
+    pub fn leader(&self) -> &TestNode {
+        &self.network().validators[0]
+    }
+
+    /// Returns the reference to test network.
+    pub fn network(&self) -> &TestNetwork {
+        &self.service_context.network
+    }
+
+    /// Returns the mutable reference to test network for manual modifications.
+    pub fn network_mut(&mut self) -> &mut TestNetwork {
+        &mut self.service_context.network
+    }
+
+    /// Returns the actual configuration of the testkit for modification.
```

Comment:
Nitpick: the description could be a little misleading; it looks like this method would return a mutable ref to `TestNetworkConfiguration` actually used by the node. I would change the description to

> Returns a copy of the actual configuration of the testkit. The returned configuration could be modified for use with `propose_configuration_change()` method.